### PR TITLE
Add Raindrops exercise

### PR DESCRIPTION
**Pull request by fwip**
I think this is the right way to add an exercise. This is my first time doing this, so if I messed anything up, please let me know. :)

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -99,6 +99,18 @@
...
+    {
+      "core": false,
+      "difficulty": 2,
+      "slug": "raindrops",
+      "topics": [
+        "control_flow_conditionals",
+        "equality",
+        "strings"
+      ],
+      "unlocked_by": null,
```
Comment (Stargator) [inline config.json]: Set "unlocked by" to anagram

File: exercises/raindrops/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+name: 'raindrops'
+dev_dependencies:
+  test: '0.12.23+1'
```
Comment (Stargator) [inline exercises/raindrops/pubspec.yaml]: Set the test version to <0.13.0 to be consistent with the other exercises

File: exercises/raindrops/test/raindrops_test.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    test("the sound for 52 is 52", () {
+      expect(raindrops.convert(52), equals("52"));
+    }, skip: true);
+    test("the sound for 105 is PlingPlangPlong as it has factors 3, 5 and 7", () {
+      expect(raindrops.convert(105), equals("PlingPlangPlong"));
+    }, skip: true);
+    test("the sound for 3125 is Plang as it has a factor 5", () {
+      expect(raindrops.convert(3215), equals("Plang"));
+    }, skip: true);
+  });
```
Comment (Stargator) [inline exercises/raindrops/test/raindrops_test.dart]: Can you add separate the tests by additional groups? Something like Anagram?

**Comment by fwip** [comment]
Sure, those all sound like good changes, I'll try to get them done today. :)

**Comment by Stargator** [comment]
@fwip Additionally, can you move the "raindrops.convert" calls to store the value in a final String variable?

Just to separate out the pieces a bit.

**Comment by Stargator** [comment]
As a final suggestion, I would add the mathematics topic to the exercise's config spec.

Move the exercise to after gigasecond and before rna-transcription. Updating rna-transcription to be unlocked by raindrops and raindrops by gigasecond.

**Comment by fwip** [comment]
Should it then be a `core` exercise, as well?

**Comment by Stargator** [comment]
yes, missed that

**Comment by Stargator** [comment]
I'm loving this, thanks again @fwip

**Comment by Stargator** [comment]
Would you mind fixing anagram so it's unlocked by rna-transcription?

Once that is done, I say it;s ready to be merged!

**Comment by fwip** [comment]
Done. :)

**Pull request merged**